src/main/java/net/fabricmc/fabric/api/client/render/FluidRenderHandler.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.render;
+
+import net.minecraft.client.render.BufferBuilder;
+import net.minecraft.client.texture.Sprite;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.ExtendedBlockView;
+
+public interface FluidRenderHandler {
+    Sprite[] getFluidSprites(ExtendedBlockView view, BlockPos pos, FluidState state);
```

Comment:
Ideally would be some way to ensure implementation is not dynamic.  (We don't want mods retrieving Sprite instances from the atlas each call.). But they *do* need to re-retrieve them on resource reload.  Maybe call this once per resource reload and cache the result in the implementation?

Reply:
>Maybe call this once per resource reload and cache the result in the implementation?

I'd make the behaviour vary depending on modder intent - I think allowing custom sprites in some conditions is fine.

Reply:
Ah, I missed that this is world-state dependent. But still, can we at least document that they shouldn't be generating this array each time?